pkg/node/node.go

```diff
@@ -175,15 +175,20 @@ func NewBee(addr string, swarmAddress swarm.Address, keystore keystore.Service,
 			return nil, err
 		}
 
-		// TODO: factory address discovery for well-known networks (goerli for beta)
-
+		var factoryAddress common.Address
 		if o.SwapFactoryAddress == "" {
-			return nil, errors.New("no known factory address")
+			var found bool
+			factoryAddress, found = chequebook.DiscoverFactoryAddress(chainID.Int64())
+			if !found {
+				return nil, errors.New("no known factory address")
+			}
 		} else if !common.IsHexAddress(o.SwapFactoryAddress) {
 			return nil, errors.New("invalid factory address")
+		} else {
+			factoryAddress = common.HexToAddress(o.SwapFactoryAddress)
```

Comment:
Log here: connect to passed in factory <address>

Reply:
introduced a log